libminifi/include/utils/StringUtils.h

```diff
@@ -105,7 +105,8 @@ class StringUtils {
    * @param s incoming string
    * @returns modified string
    */
-  static inline std::string trimLeft(std::string s) {
+  static inline std::string trimLeft(const std::string &in_string) {
+    auto s = in_string;
```

Comment:
Ah, you just replied it between I wrote it and submitted it. Thanks!

Reply:
ha I guess this didn't come through: 
https://github.com/apache/nifi-minifi-cpp/commit/809a46dbbe91b765b0b85d11ae8a6c2ed6f1751b#r34088401

I'm reverting this because it is useless to the PR and was intended for another branch ( was git stashing like a mad man helping someone ).

Reply:
fwiw I was testing was optimizations on godbolt of varying compilers with varying string lengths ( to impose SSO ) and compilers to look at how string interning would impact the resulting assembler output. Frankly even though I "preferred" the semantics of the prototype I didn't change the functional semantics and actually made it worse at optimization levels < 2! I didn't intend to include but will still run the various tests I ran just to be cautious and remove those sleeps!